src/projectedTurnout/projectedTurnout.controller.ts

```diff
@@ -12,8 +12,59 @@ export class ProjectedTurnoutController {
   async getProjectedTurnout(@Query() dto: ProjectedTurnoutUniqueDTO) {
     const record = await this.projectedTurnoutService.getProjectedTurnout(dto)
     if (!record) {
-      throw new NotFoundException(`Projected turnout not found for ${dto}`)
+      // Format the error message in a more readable way
+      const formattedParams = this.formatDtoForError(dto)
+      throw new NotFoundException(
+        `Projected turnout not found for the following parameters:\n${formattedParams}`,
+      )
     }
     return record
   }
+
+  private formatDtoForError(dto: ProjectedTurnoutUniqueDTO): string {
+    const params: string[] = []
+
+    if (dto.state) {
+      params.push(`State: ${dto.state}`)
+    }
+
+    if (dto.electionDate) {
+      params.push(`Election Date: ${dto.electionDate}`)
+    }
+
+    if (dto.L2DistrictType) {
+      let districtTypeLabel = ''
+      if (typeof dto.L2DistrictType === 'string') {
+        try {
+          const parsed = JSON.parse(dto.L2DistrictType)
+          districtTypeLabel =
+            parsed.label || parsed.L2DistrictType || dto.L2DistrictType
+        } catch {
+          // If parsing fails, use the string as-is
+          districtTypeLabel = dto.L2DistrictType
+        }
+      } else {
+        districtTypeLabel = String(dto.L2DistrictType)
+      }
+      params.push(`District Type: ${districtTypeLabel}`)
+    }
+
+    if (dto.L2DistrictName) {
+      let districtName = ''
+      if (typeof dto.L2DistrictName === 'string') {
+        try {
+          const parsed = JSON.parse(dto.L2DistrictName)
+          districtName = parsed.L2DistrictName || dto.L2DistrictName
+        } catch {
+          // If parsing fails, use the string as-is
+          districtName = dto.L2DistrictName
+        }
+      } else {
+        districtName = String(dto.L2DistrictName)
+      }
+      params.push(`District Name: ${districtName}`)
+    }
+
+    return params.join('\n')
```

Comment:
Each `params.push` here could easily be abstracted as something like:

```typescript
addParam(label: string, value:string)
```

That way instead of a method full of conditionals, you can just programatically handle the usecases.

You could also abstract the handling of the two  fields that require JSON parsing, further simplifying the cyclomatic complexity of this method.